physics/discrete_trajectory_body.hpp

```diff
@@ -158,6 +160,44 @@ void DiscreteTrajectory<Frame>::Append(
   CHECK(--timeline_.end() == it)
       << "Append out of order at " << time << ", last time is "
       << (--timeline_.end())->first;
+  if (downsampling_.has_value()) {
+    if (timeline_.size() == 1) {
+      downsampling_->start_of_dense_timeline = timeline_.begin();
+      downsampling_->dense_intervals = 0;
+    } else {
+      CheckNoForksBefore(last().time());
+      ++downsampling_->dense_intervals;
+      if (downsampling_->dense_intervals ==
+          downsampling_->max_dense_intervals) {
+        std::vector<TimelineConstIterator> dense_iterators;
+        dense_iterators.reserve(downsampling_->max_dense_intervals + 1);
```

Comment:
Comment the `+ 1`: this is a number of points.

Reply:
Done.